### ISSUE TITLE
Trinity does not attempt to create directories

### ISSUE THREAD

**Issue by MysticRyuujin** [ethereum/py-evm#922]
So does Trinity also not create log files? 😄 

```
root@trinity01:~# ls /root/trinity/
logs
root@trinity01:~# trinity --port 30304 --data-dir /root/trinity
   ERROR  06-17 08:21:50        main  
It appears that /root/trinity/logs/trinity.log does not exist.
Trinity does not attempt to create directories outside of its root path
Either manually create the path or ensure you are using a data directory
inside the XDG_TRINITY_ROOT path
```

**Comment by MysticRyuujin** [comment]
So obviously I need to set `--trinity-root-dir` but I think this error is still relevant

**Comment by cburgdorf** [comment]
As long as the directory `/root/trinity/logs/` exists, that *should* work fine **on latest master** (was fixed by https://github.com/ethereum/py-evm/commit/4958c6ab9ba8819048c99877675d3da98a110326).

Can you double check if that works on latest master if the directory `/root/trinity/logs/` exists?

**Comment by MysticRyuujin** [comment]
@cburgdorf I just recently (like 30 minutes ago) did a `pip3 install -U trinity` so I have whatever it most recently released?

See the top line of my output, the folder exists

```
root@trinity01:~# ls /root/trinity/
data  logs
root@trinity01:~# trinity --port 30304 --trinity-root-dir /root/trinity --data-dir /root/trinity/data
   ERROR  06-17 08:38:38        main  
It appears that /root/trinity/data/logs/trinity.log does not exist.
Trinity does not attempt to create directories outside of its root path
Either manually create the path or ensure you are using a data directory
inside the XDG_TRINITY_ROOT path
```

**Comment by MysticRyuujin** [comment]
@cburgdorf if you've got quick instructions how to update the pip3 installed version to the github master version I'd be happy to run / /test it (Ubuntu)

**Comment by cburgdorf** [comment]
I just checked the latest release and it didn't have the fix for this issue in yet. We plan to cut a new release this week though.

If you like to get your node on latest master, just follow along the [contributing guide](http://py-evm.readthedocs.io/en/latest/contributing.html).

**Comment by carver** [comment]
Although the fix isn't released yet, also note that the logs are going inside the `--data-dir`, so you would have to manually create the folder `/root/trinity/data/logs` before the above `trinity` runner. (instead of `/root/trinity/logs`)

**Comment by pipermerriam** [comment]
@carver I think that warrants us revisiting the logic on this.  *if* `--data-dir` is **not** under the `TRINITY_HOME` dir, but it does exist, it seems safe to create directories there (same with any other directory structure that we would create).

**Comment by carver** [comment]
I think that's ok. I'm actually not sure what specific issue we're trying to prevent by not creating directories, even inside TRINITY_HOME. Just to catch if someone accidentally specified an incorrect `data-dir`?

**Comment by pipermerriam** [comment]
fixed by #1268

**Comment by pipermerriam** [comment]
not quite merged, got confused.
